pkgs/by-name/sp/space-cadet-pinball/package.nix

```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchzip,
+  cmake,
+  SDL2,
+  SDL2_mixer,
+  unrar,
+}:
+
+let
+  assets = fetchzip {
+    url = "https://archive.org/download/SpaceCadet_Plus95/Space_Cadet.rar";
+    hash = "sha256-fC+zsR8BY6vXpUkVd6i1jF0IZZxVKVvNi6VWCKT+pA4=";
+    stripRoot = false;
+    nativeBuildInputs = [ unrar ];
+  };
+  darwinApp = "$out/Applications/SpaceCadetPinball.app/Contents";
+  assetsDest =
+    if stdenv.hostPlatform.isDarwin then darwinApp + "/Resources" else "$out/share/SpaceCadetPinball";
+in
+stdenv.mkDerivation rec {
+  pname = "SpaceCadetPinball";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner = "k4zmu2a";
+    repo = "SpaceCadetPinball";
+    rev = "Release_${version}";
+    hash = "sha256-W2P7Txv3RtmKhQ5c0+b4ghf+OMsN+ydUZt+6tB+LClM=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [
+    SDL2
+    SDL2_mixer
+  ];
+
+  postPatch = ''
+    # Change the hardcoded FHS assets path
+    substituteInPlace SpaceCadetPinball/pch.h \
+      --replace-fail /usr/share ${placeholder "out"}/share
+    # Disable building a universal binary on Darwin, otherwise the cc wrapper passing -arch breaks the build
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "arm64;x86_64" ""
+  '';
+
+  # Darwin needs a custom installPhase since it is excluded from the cmake install
+  # https://github.com/k4zmu2a/SpaceCadetPinball/blob/0f88e43ba261bc21fa5c3ef9d44969a2a079d0de/CMakeLists.txt#L221
+  # This builds a bundle similar to what upstream's build script produces
+  # https://github.com/k4zmu2a/SpaceCadetPinball/blob/cb9b7b886244a27773f66b0b19fdc2998392565e/build-mac-app.sh
+  installPhase = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    runHook preInstall
+    install -D ../bin/SpaceCadetPinball -t ${darwinApp}/MacOS
+    install -Dm644 ../Platform/macOS/SpaceCadetPinball.icns -t ${darwinApp}/Resources
+    substitute ../Platform/macOS/Info.plist ${darwinApp}/Info.plist \
+      --replace-fail CHANGEME_SW_VERSION ${version}
+    echo -n "APPL????" > ${darwinApp}/PkgInfo
+    runHook postInstall
+  '';
```

Comment:
This is an annoying quirk of how mkDerivation works here but this will not execute the hooks on non-Darwin because the installPhase will be an empty string there.

Are you sure the postInstall below actually runs?

Reply:
As far as I can tell having built it in a NixOS VM just now, the postInstall does run as the assets seem to be copied to the output path as they should. Don't know if this is only a problem in some cases though?